.github/workflows/main-build-windows.yml

```diff
@@ -19,6 +19,11 @@ jobs:
     steps:
       - uses: actions/checkout@v4
 
+      - name: Initialize CodeQL
+        uses: github/codeql-action/init@v2
+        with:
+          languages: cpp, python

```

Comment:
similar comment as above.

Do github workflows have any concept of include files/macros?  Seems like this is redundant in several files being modified in this PR, it would be nice to have a single place to define your changes, e.g. if you remove "python" and it needs to be added back later.

Reply:
Yes, Github has a concept of custom "Actions" that can be defined in a file and then re-used across different workflows. Making that change now